alchemist/alchemist-fxui/src/main/java/it/unibo/alchemist/boundary/gui/effects/json/EffectSerializer.java

```diff
@@ -0,0 +1,387 @@
+        final T deserialized = GSON.fromJson(reader, type.getType());
+        reader.close();
+        return deserialized;
+    }
+
+    /**
+     * Saves a generic file with a generic writer and closes that writer.
+     *
+     * @param writer a specified writer to use
+     * @param object the object to serialize
+     * @param type   a specified type token to get right type from
+     * @param <T>    the generic type of object to save
+     * @throws com.google.gson.JsonIOException If there was a problem writing to the writer
+     * @throws IOException     If some other I/O error occurs
+     */
+    private static <T> void save(final Writer writer, final T object, final TypeToken<T> type) throws IOException {
+        GSON.toJson(object, type.getType(), writer);
+        writer.close();
+    }
+
+    /**
+     * Get an {@link EffectFX Effect} from the specified file. It tries to
+     * deserialize a JSON file.
+     *
+     * @param <P> the position type
+     * @param effectFile Source file
+     * @return Effect loaded from the file
+     * @throws java.io.FileNotFoundException If the file does not exist, is a directory rather than a
+     *                               regular file, or for some other reason cannot be opened for
+     *                               reading
+     * @throws com.google.gson.JsonIOException       If there was a problem reading from the Reader
+     * @throws com.google.gson.JsonSyntaxException   If JSON is not a valid representation for an object of type
+     * @throws IOException           If some other I/O error occurs
+     */
+    public static <P extends Position2D<? extends P>> EffectFX<P> effectFromFile(
+            final File effectFile
+    ) throws IOException {
+        return load(new FileReader(effectFile, DEFAULT_CHARSET), new TypeToken<>() { });
+    }
+
+    /**
+     * Get an {@link EffectFX Effect} from the specified resource file. It tries to
+     * deserialize a JSON file.
+     *
+     * @param <P> the position type
+     * @param resource resource file
+     * @return Effect loaded from the resource file
+     * @throws java.io.FileNotFoundException If the file does not exist, is a directory rather than a
+     *                               regular file, or for some other reason cannot be opened for
+     *                               reading
+     * @throws com.google.gson.JsonIOException       If there was a problem reading from the Reader
+     * @throws com.google.gson.JsonSyntaxException   If JSON is not a valid representation for an object of type
+     * @throws IOException           If some other I/O error occurs
+     */
+    public static <P extends Position2D<? extends P>> EffectFX<P> effectFromResources(
+            final String resource
+    ) throws IOException {
+        return load(
+                new InputStreamReader(ResourceLoader.getResourceAsStream(resource), DEFAULT_CHARSET),
+                new TypeToken<>() { }
+                );
```

Comment:
Indentation is strange here, you should indent the parenthesis one less level maybe